DomainContent/LoadTest/RaffleWheel/raffleWheel.js

```diff
@@ -0,0 +1,138 @@
+// raffleWheel.js
+// 
+// Created by Rebecca Stankus on 07/05/2018
+// Copyright High Fidelity 2018
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+//
+/* global Avatar, AccountServices */
+
+(function() {
+    var ANGULAR_VELOCITY = {
+        x: 10,
+        y: 0,
+        z: 0
+    };
+    var ANGULAR_VELOCITY_CHECK_MS = 50;
+    var ANGULAR_VELOCITY_DECREMENT = 0.09;
+    var CHECKING_INTERVAL_DELAY = 100;
+    var USERS_ALLOWED_TO_SPIN_WHEEL = ['philip', 'ryan', 'Becky', 'MissLiviRose', 'Firebird25', 'Alan_'];
+    var MAX_CHAR_PER_LINE = 9;
+    var CUT_OFF_1 = 13;
+    var CUT_OFF_2 = 18;
+    var CUT_OFF_3 = 24;
+    var CUT_OFF_4 = 32;
+    var LINE_HEIGHT_1 =0.21;
+    var LINE_HEIGHT_2 =0.15;
+    var LINE_HEIGHT_3 =0.115;
+    var LINE_HEIGHT_4 =0.075;
+    var DEFAULT_LINE_HEIGHT = 0.3;
+
+
+    var _this;
+
+    function shuffle(array) {
+        var currentIndex = array.length, temporaryValue, randomIndex;
+      
+        while (0 !== currentIndex) {
+            randomIndex = Math.floor(Math.random() * currentIndex);
+            currentIndex -= 1;
+            temporaryValue = array[currentIndex];
+            array[currentIndex] = array[randomIndex];
+            array[randomIndex] = temporaryValue;
+        }
+        return array;
+    }
+
+    var Wheel = function() {
+        _this = this;
+    };
+
+    Wheel.prototype = {
+        interval: null,
+        angularVelocityLimit: 10,
+        nameText: null,
+        avatarList: [],
+        usernameList: [],
+        listCounter: 0,
+
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            Users.usernameFromIDReply.connect(_this.usernameFromIDReply);
+        },
+
+        mousePressOnEntity: function(entityID, mouseEvent) {
+            if (!mouseEvent.button === "Primary") {
+                return;
+            }
+            var position = Entities.getEntityProperties(_this.entityID, 'position').position;
+            var winnerTextEntitiesArray = Entities.findEntitiesByName("Raffle Wheel Winner", position, 1);
+            _this.usernameList = [];
+            if (USERS_ALLOWED_TO_SPIN_WHEEL.indexOf(AccountServices.username) >= 0) {
+                _this.angularVelocityLimit = 10;
+                _this.avatarList = AvatarList.getAvatarIdentifiers();
+                _this.avatarList.forEach(function(id) {
+                    Users.requestUsernameFromID(id);
+                });
+                Entities.editEntity(_this.entityID, {
+                    angularVelocity: ANGULAR_VELOCITY
+                });
+                Script.setTimeout(function() {
+                    if (_this.interval) {
+                        Script.clearInterval(_this.interval);
+                    }
+                    var winner;
+                    _this.interval = Script.setInterval(function() {
+                        var lineHeight;
+                        shuffle(_this.usernameList);
```

Comment:
Can you just shuffle once before spinning? It looks like this is shuffling each time the user account ticks